front/components/assistant_builder/InstructionScreen.tsx

```diff
@@ -353,130 +357,122 @@ function Suggestions({
 
     if (!instructions.trim()) {
       setError(null);
-      setLoading(false);
-      setSuggestions(STATIC_SUGGESTIONS);
+      setSuggestionsStatus(
+        suggestions.length > 0 ? "suggestions_available" : "no_suggestions"
+      );
       clearTimeout(debounceHandle.current);
       return;
     }
 
     const updateSuggestions = async () => {
-      setLoading(true);
-      const updatedSuggestions = await getUpdatedInstructionsSuggestions({
+      setSuggestionsStatus("loading");
+      const updatedSuggestions = await getRankedSuggestions({
         owner,
         currentInstructions: instructions,
-        formerSuggestions:
-          suggestions.status === "ok"
-            ? suggestions.suggestions.slice(0, 2)
-            : [],
+        formerSuggestions: suggestions.slice(0, 2),
       });
       if (updatedSuggestions.isErr()) {
         setError(updatedSuggestions.error);
-        setLoading(false);
+        setSuggestionsStatus("error");
         return;
       }
-
-      setSuggestions(updatedSuggestions.value);
+      if (
+        updatedSuggestions.value.status === "ok" &&
+        !updatedSuggestions.value.suggestions.length
+      ) {
+        setSuggestionsStatus("instructions_are_good");
```

Comment:
Shouldn't we return here? It looks like we will set `instructions_are_good` and then right after update to `suggestions_available` 🤔